src/FileIndexer.ts

```diff
@@ -124,12 +125,47 @@ export class FileIndexer {
       if (isDefinition) {
         this.addSymbolInformation(node, sym, declaration, lsifSymbol)
         this.handleShorthandPropertyDefinition(declaration, range)
+        this.handleObjectBindingPattern(node, range)
         // Only emit one symbol for definitions sites, see https://github.com/sourcegraph/lsif-typescript/issues/45
         break
       }
     }
   }
 
+  /**
+   * Emits an additional definition occurrence when destructuring an object
+   * pattern. For example:
+   * ```
+   * interface Props { property: number}
+   * const props: Props[] = [{ property: 42 }]
+   * props.map(({property}) => property) = {a}
+   * //          ^^^^^^^^ references `Props.property` and defines a local parameter `property`
+   * ```
+   */
+  private handleObjectBindingPattern(node: ts.Node, range: number[]): void {
+    if (
+      !ts.isIdentifier(node) ||
+      !ts.isBindingElement(node.parent) ||
+      !ts.isObjectBindingPattern(node.parent.parent)
+    ) {
+      return
+    }
```

Comment:
```suggestion
    const isBinding = 
          ts.isIdentifier(node) &&
          ts.isBindingElement(node.parent) &&
          ts.isObjectBindingPattern(node.parent.parent)
    if (!isBinding) {
      return
    }
```
nit: I think having a single `!` and assigning a name would make things more readable.

Reply:
Fair enough. Done